filepath.cabal

```diff
@@ -48,20 +48,19 @@ library
         System.FilePath.Windows
 
     build-depends:
-        base >= 4 && < 4.15
+        base >= 4.3 && < 4.15
 
     ghc-options: -Wall
 
 test-suite filepath-tests
     type: exitcode-stdio-1.0
     default-language: Haskell2010
     main-is: Test.hs
-    ghc-options: -main-is Test
     hs-source-dirs: tests
     other-modules:
         TestGen
         TestUtil
     build-depends:
         filepath,
         base,
-        QuickCheck >= 2.7 && < 2.14
+        QuickCheck >= 2.14 && < 2.15
```

Comment:
Why is this a good thing? We demand people to upgrade QuickCheck and a whole pile of other dependencies for what benefit?

Reply:
I am also confused by this change.

Reply:
This is a test dependency. `QuickCheck` is one of the lightest packages you can imagine, and there isn't even any `tasty-quickcheck` or similar package to recompile after it. QuickCheck-2.10 (released in 2017) made

> Random strings and characters now include Unicode characters 

change, which for example broke `bytestring` test-suite. Allowing older `QuickCheck` thus introduces more variation in what kind of tests are actually run. Take that to the limit, and using single `QuickCheck` major series should be enough.

Specially note, that e.g. Stackage is not upgrading `filepath`, so their setup is not running `filepath` tests. Neither does GHC run boot libraries test suites.

In other words, this change only affects `filepath` CI (which is fine), and direct developers (which there are a handful only).

Tell me a good reason to allow building `filepath` tests with QuickCheck-2.7?